tests/BaseSFFS.device.nut

```diff
@@ -0,0 +1,110 @@
+// MIT License
+//
+// Copyright 2017 Electric Imp
+//
+// SPDX-License-Identifier: MIT
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO
+// EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES
+// OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+// ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+// "Promise" symbol is injected dependency from ImpUnit_Promise module,
+// while class being tested can be accessed from global scope as "::Promise".
+
+// Base class for the SFFS test cases
+// to share common functionality
+class BaseSffsTestCase extends ImpTestCase {
+
+  size = 0;
+  sffs = null;
+  pages = 0;
+  start = 0;
+
+  //
+  // Initialization of the SpiFlashFileSystem library
+  //  - create an SpiFlashFileSystem class object
+  //  - erase all SPI flash data
+  //  - initialize a new FAT block
+  function setUp() {
+    return Promise(function(ok, err) {
+      // check that we're on  003+
+
+      this.assertTrue("spiflash" in hardware, "imp003 and above is expected");
+
+      // get actual flash size
+      hardware.spiflash.enable();
+      this.size = hardware.spiflash.size();
```

Comment:
Can I change "this.size" on "size" on the callback methods, when I make bindenv(this) for a callback method ?